lib/apiservers/engine/proxy/archive.go

```diff
@@ -32,7 +32,7 @@ import (
 
 type VicArchiveProxy interface {
 	ArchiveExportReader(ctx context.Context, store, ancestorStore, deviceID, ancestor string, data bool, filterSpec vicarchive.FilterSpec) (io.ReadCloser, error)
-	ArchiveImportWriter(ctx context.Context, store, deviceID string, filterSpec vicarchive.FilterSpec) (io.WriteCloser, error)
+	ArchiveImportWriter(ctx context.Context, store, deviceID string, filterSpec vicarchive.FilterSpec, errchan chan error) (io.WriteCloser, error)
```

Comment:
Already commented above.  I usually see something like this,

ArchiveImportWriter(ctx context.Context, store, deviceID string, filterSpec vicarchive.FilterSpec) (io.WriteCloser, <-
chan error, error)

The <- indicates it's for reading only.  Not sure if returning it rather than passing in via arg is idiomatic.  Not a big deal.